Mathlib/Data/Finset/NatAntidiagonal.lean

```diff
@@ -28,19 +34,21 @@ namespace Nat
 
 /-- The antidiagonal of a natural number `n` is
     the finset of pairs `(i, j)` such that `i + j = n`. -/
-def antidiagonal (n : ℕ) : Finset (ℕ × ℕ) :=
-  ⟨Multiset.Nat.antidiagonal n, Multiset.Nat.nodup_antidiagonal n⟩
-#align finset.nat.antidiagonal Finset.Nat.antidiagonal
-
-/-- A pair (i, j) is contained in the antidiagonal of `n` if and only if `i + j = n`. -/
-@[simp]
-theorem mem_antidiagonal {n : ℕ} {x : ℕ × ℕ} : x ∈ antidiagonal n ↔ x.1 + x.2 = n := by
-  rw [antidiagonal, mem_def, Multiset.Nat.mem_antidiagonal]
-#align finset.nat.mem_antidiagonal Finset.Nat.mem_antidiagonal
+instance : HasAntidiagonal ℕ where
+  antidiagonal n := ⟨Multiset.Nat.antidiagonal n, Multiset.Nat.nodup_antidiagonal n⟩
+  mem_antidiagonal n xy := by
+    rw [mem_def, Multiset.Nat.mem_antidiagonal]
+
+/-- `Finset.Nat.antidiagonal` coincides with `Finset.antidiagonalOfLocallyFinite.antidiagonal` -/
+lemma antidiagonal_eq_antidiagonal (n : ℕ):
+    antidiagonal n = Finset.antidiagonalOfLocallyFinite.antidiagonal n := by
+    suffices : Finset.Nat.instHasAntidiagonalNatAddMonoid = Finset.antidiagonalOfLocallyFinite
+    rw [← this]
+    apply Subsingleton.elim
```

Comment:
This can now be removed entirely, I think the subsingleton instance alone is enough.